src/Adapter/Image/ImageRetriever.php

```diff
@@ -353,22 +353,44 @@ public function getCustomizationImage($imageHash)
     public function getNoPictureImage(Language $language)
     {
         $urls = [];
-        $type = 'products';
-        $imageTypes = ImageType::getImagesTypes($type, true);
 
-        if (empty($imageTypes)) {
-            throw new PrestaShopException(sprintf('There is no image type defined for "%s".', $type));
-        }
+        // Get all image types present on shops
+        $imageTypes = ImageType::getImagesTypes();
+
+        // Get path of original image for our current language
+        $originalImagePath = implode(DIRECTORY_SEPARATOR, [
```

Comment:
They are present from the installer for all languages, they must be there.
In `_PS_CAT_IMG_DIR_` `_PS_MANU_IMG_DIR_` `_PS_SUPP_IMG_DIR_` `_PS_PRODUCT_IMG_DIR_` `_PS_STORE_IMG_DIR_`.

In ImageThumbnailsRegenerator that is used in backoffice, there is a following fallback, but it doesn't make much sense. Use `_PS_PRODUCT_IMG_DIR_` if the image is not in `_PS_CAT_IMG_DIR_`?

```php
$file = $dir . $language->getIsoCode() . '.jpg';
                if (!file_exists($file)) {
                    $file = _PS_PRODUCT_IMG_DIR_ . $defaultLang->getIsoCode() . '.jpg';
                }
```

We could introduce one global fallback image for these cases, or when user creates a new completely unknown language. Or just use an icon, this would simplify everything. :-)

Reply:
I introduced a fallback that is used now.